java-checks/src/test/files/checks/spring/SpringAntMatcherOrderCheck.java

```diff
@@ -0,0 +1,69 @@
+package files.checks.spring;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.AbstractRequestMatcherRegistry;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configurers.ExpressionUrlAuthorizationConfigurer;
+
+public class App {
+
+  protected void configure(HttpSecurity http, String dynamicUrl) throws Exception {
+
+    http
+      .authorizeRequests()
+        .antMatchers("/login/*me").permitAll()
+        .antMatchers("/login").hasRole("ADMIN")
+        .antMatchers("/login/home").permitAll(); // Noncompliant [[sc=22;ec=35;secondary=14]] {{Reorder the URL patterns from most to less specific, the pattern "/login/home" should occurs before "/login/*me".}}
+
+    http
+      .authorizeRequests()
+      .antMatchers(
+        "/lo?in",
+        dynamicUrl,
+        "/login" // Noncompliant [[sc=9;ec=17;secondary=21]] {{Reorder the URL patterns from most to less specific, the pattern "/login" should occurs before "/lo?in".}}
+      ).permitAll();
+
+    http
+      .authorizeRequests()
+        .antMatchers("/login/home").permitAll()
+        .antMatchers("/login").hasRole("ADMIN")
+        .antMatchers("/login/*me").permitAll();
+
+    http
+      .authorizeRequests()
+      .antMatchers(HttpMethod.GET,"/home").permitAll()
+      .antMatchers(HttpMethod.POST,"/home").permitAll();
+
+    http
+      .antMatcher("/root1")
+        .authorizeRequests()
+          .antMatchers("/level1").permitAll()
+    .and()
+      .antMatcher("/root2")
+        .authorizeRequests()
+          .antMatchers("/level1").permitAll();
+
+    http
+      .antMatcher("/root1")
+        .authorizeRequests()
+          .antMatchers("/level1").permitAll()
+          .antMatchers("/level1").permitAll(); // Noncompliant
+
+    http
```

Comment:
As discussed, I would appreciate to see a test case with more than one issue on a sequence of `antMatchers`, like in the example from the rule documentation.

Reply:
I forgot this case in my "Fix from review" commit. I'm going to add a small new commit.